microsoft-edge/devtools-guide-chromium/experimental-features/index.md

```diff
@@ -641,52 +611,87 @@ See also:
+Distinguishes postMessage events from other timeline events, to easily investigate performance issues related to post messages across various threads in applications.
 
-![The Autofill tool](./index-images/autofill-tool.png)
+This experiment improves the **Performance** tool's timeline to help you quickly identify `postMessage` events and handlers, by distinguishing `postMessage` events from other timeline events.  Without this experiment, `postMessage` dispatch events and handler events both appear as generic scripting function call events.  With this experiment enabled:
+* `postMessage` dispatch events appear on the timeline as **Schedule postMessage**.
+* `postMessage` handler events appear on the timeline as **On Message**.
+
+!["Schedule postMessage" events and "On Message" events in the Performance tool](./index-images/post-message-events.png)
+
+The performance timeline thus helps you investigate when a `postMessage` call occurred, and how long the message was queued before the handler starts.  The dispatch events are linked to handler events with arrow initiators that appear when you click on either type of event:
+
+![Arrows linking dispatch events to handler events](./index-images/post-message-events-arrows.png)
 
 See also:
-* [Autofill](../../privacy-whitepaper/index.md#autofill) in _Microsoft Edge Privacy Whitepaper_.
-* [Autofill](../../webview2/concepts/overview-features-apis.md?tabs=dotnetcsharp#autofill) in _Overview of WebView2 features and APIs_.
+* [Timeline event reference](/microsoft-edge/devtools-guide-chromium/evaluate-performance/performance-reference)
+* [PostMessage Trace Events demo (rendered)](https://microsoftedge.github.io/Demos/devtools-postmessage-perf-timeline/)
+* [PostMessage Trace Events demo (source code)](https://github.com/MicrosoftEdge/Demos/tree/main/devtools-postmessage-perf-timeline)
+<!-- * [postMessage events are distinguished from other timeline events](https://review.learn.microsoft.com/en-us/microsoft-edge/devtools-guide-chromium/whats-new/2024/04/devtools-124?branch=pr-en-us-3129#postmessage-events-are-distinguished-from-other-timeline-events) in _What's New in DevTools (Microsoft Edge 124)_. -->
 
 Status:
-* This checkbox is present in Microsoft Edge Canary 122.
-* This checkbox is not present in Microsoft Edge Stable 120.
+* This checkbox is present in Microsoft Edge Canary 125.
+* This checkbox is not present in Microsoft Edge Stable 123.
+
+
+<!-- ====================================================================== -->
+## Enable save and load trace with annotations in Performance Panel
+<!-- no (?) link -->
+
+_new as of Microsoft Edge 125_
+
+Enables save and load trace with annotations in the **Performance** tool.
+<!-- todo: need desc -->
```

Comment:
Enable save and load trace with annotations in Performance Panel - need real description, if possible.
@captainbrosset

Reply:
I couldn't find anything about this. I will ask Rob to help if he can. The bug that introduced this in Chromium was done by the Chrome team, and is unfortunately not publicly visible.

Reply:
Trying to find out more about this experiment. Will report when I know more.

Reply:
The "minimap-driven navigation" (aka "breadcrumbs", a feature that enables developers to crop into and focus on the more relevant parts of a trace, see WNIDT 122:
https://developer.chrome.com/blog/new-in-devtools-122?hl=en#perf-breadcrumbs)
and the "track data context-menu" (a feature that enables developers to merge and hide superfluous or irrelevant entries of a trace, cleaning up the flame charts and trees, see WNIDT 124:
https://developer.chrome.com/blog/new-in-devtools-124#hide-func)
This experiment added the ability to persist those inside trace files, and reapply them when importing traces that contain breadcrumbs and altered tracks.

from @captainbrosset offline